.github/workflows/scripts/bump-go.sh

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env bash
+#
+# bump-go.sh — Update go.mod `go` directive and toolchain to latest stable Go release.
+#
+# Usage:
+#   ./bump-go.sh [--apply|-a] <path/to/go.mod>
+#
+# By default the script runs in *dry‑run* mode: it creates a local branch,
+# commits the version bump, shows the exact patch, **checks for an existing PR**
+# with the same title, and exits. Nothing is pushed. The temporary branch is
+# deleted automatically on exit, so your working tree stays clean. Pass
+# --apply (or -a) to push the branch and open a new PR *only if one doesn’t
+# already exist*.
+# -----------------------------------------------------------------------------
+set -euo pipefail
+
+usage() {
+  echo "Usage: $0 [--apply|-a] <path/to/go.mod>" >&2
+  exit 1
+}
+
+require_cmd() {
+  command -v "$1" >/dev/null 2>&1 || {
+    echo "Error: '$1' is required but was not found in PATH" >&2
+    exit 1
+  }
+}
+
+# ---- Preconditions ----------------------------------------------------------
+for cmd in curl jq git gh; do
+  require_cmd "$cmd"
+done
+
+# ---- Argument parsing -------------------------------------------------------
+APPLY=0
+GO_MOD=""
+
+while [[ $# -gt 0 ]]; do
+  case "$1" in
+    --apply|-a) APPLY=1 ;;
+    -h|--help)  usage ;;
+    *)          [[ -z "$GO_MOD" ]] && GO_MOD="$1" || usage ;;
+  esac
+  shift
+done
+
+[[ -z "$GO_MOD" ]] && usage
+[[ -f "$GO_MOD" ]] || { echo "Error: '$GO_MOD' not found" >&2; exit 1; }
+
+# ---- Discover latest stable Go release --------------------------------------
+echo "Fetching latest stable Go version…"
+LATEST_JSON=$(curl -fsSL https://go.dev/dl/?mode=json | jq -c '[.[] | select(.stable==true)][0]')
+FULL_VERSION=$(jq -r '.version' <<< "$LATEST_JSON")        # e.g. go1.23.4
+TOOLCHAIN_VERSION="${FULL_VERSION#go}"                     # e.g. 1.23.4
+GO_DIRECTIVE_VERSION=$(cut -d. -f1-2 <<< "$TOOLCHAIN_VERSION")
+
+echo "  → go : $GO_DIRECTIVE_VERSION"
+echo "  → toolchain : $TOOLCHAIN_VERSION"
+
+# ---- Prepare Git branch ---------------------------------------------------
+CURRENT_GO_DIRECTIVE=$(grep -E '^go ' "$GO_MOD" | awk '{print $2}')
+CURRENT_TOOLCHAIN_DIRECTIVE=$(grep -E '^toolchain ' "$GO_MOD" | awk '{print $2}' | cut -d' ' -f2)
```

Comment:
The `cut` in this pipe is redundant. However, if I were to pick one, I'd go with `cut` since it's easier to read than `awk` commands. There's also one in the line above this one.